### PR TITLE
Allow the use of ADC2 with a flag.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Comment by floitsch** [comment]
I will write a test, if you agree with this approach.

File: src/resources/adc_esp32.cc [hunk 1]
```diff
@@ -58,6 +55,75 @@ static int get_adc2_channel(int pin) {
...
+    case 20: return ADC2_CHANNEL_9;
+    default: return adc2_channel_t(-1);
+  }
+}
+
+#elif CONFIG_IDF_TARGET_ESP32C3
+
+static int get_adc1_channel(int pin) {
+  switch (pin) {
+    case 0: return ADC1_CHANNEL_0;
```
Comment (floitsch) [inline src/resources/adc_esp32.cc:96]: Note that the pin numbers for the ESP32C2 were wrong.

File: lib/gpio/adc.toit [hunk 2]
```diff
@@ -42,12 +44,18 @@ class Adc:
   Note that chip-specific limitations apply, generally the precision at
     various voltage ranges.
 
-  On the ESP32, only the ADC1 (pins 32-39) is supported. The ADC2 has too many
-    restrictions (cannot be used when WiFi is active, and some of the pins are
-    strapping pins), and is therefore disabled.
+  If $allow_dangerous is provided, actives ADC pins that are dangerous to use.
```
Comment (kasperl) [inline lib/gpio/adc.toit:47]: Is it really dangerous or just unreliable, unstable, restricted, special-purpose?

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Comment by floitsch** [comment]
Still need to get the test running...

**Pull request merged**